### DISCUSSION TITLE
UNABLE TO ACTIVATE APEX LSP

### DISCUSSION THREAD

**Discussion by vikrambagali** [forcedotcom/salesforcedx-vscode#5323] [Q&A]
[{
	"resource": "/ApexLSP",
	"owner": "apex",
	"severity": 8,
	"message": "Unable to activate the Apex Language Server - The salesforcedx-vscode-apex.java.home setting defined in VS Code settings points to a missing folder. For information on how to setup the Salesforce Apex extension, see [Set Your Java Version](https://developer.salesforce.com/tools/vscode/en/vscode-desktop/java-setup).",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]

**Comment by vikrambagali** [comment]
Give me a solution for this. 
i tried all the ways but i did not get the  results.
re-installed the jdk new version old and all versions
and pasted the file at correct path in apex home, settings- json.

**Comment by peternhale** [comment]
@vikrambagali would you share the text from the `Apex Language Server` output channel? There may be additional message there to help determine the cause of this issue.
